doc/modules/svm.rst

```diff
@@ -519,77 +551,78 @@ instance that will use that kernel::
-method. At the moment, the kernel values between *all* training vectors and the
-test vectors must be provided.
+You can pass pre-computed kernels by using the ``kernel='precomputed'``
+option. You should then pass Gram matrix instead of X to the `fit` and
+`predict` methods. The kernel values between *all* training vectors and the
+test vectors must be provided:
 
     >>> import numpy as np
+    >>> from sklearn.datasets import make_classification
+    >>> from sklearn.model_selection import train_test_split 
     >>> from sklearn import svm
-    >>> X = np.array([[0, 0], [1, 1]])
-    >>> y = [0, 1]
+    >>> X, y = make_classification(n_samples=10, random_state=0)
+    >>> X_train , X_test , y_train, y_test = train_test_split(X, y, random_state=0)
     >>> clf = svm.SVC(kernel='precomputed')
     >>> # linear kernel computation
-    >>> gram = np.dot(X, X.T)
-    >>> clf.fit(gram, y)
+    >>> gram_train = np.dot(X_train, X_train.T)
+    >>> clf.fit(gram_train, y_train)
     SVC(kernel='precomputed')
     >>> # predict on training examples
-    >>> clf.predict(gram)
-    array([0, 1])
-
-Parameters of the RBF Kernel
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-When training an SVM with the *Radial Basis Function* (RBF) kernel, two
-parameters must be considered: ``C`` and ``gamma``.  The parameter ``C``,
-common to all SVM kernels, trades off misclassification of training examples
-against simplicity of the decision surface. A low ``C`` makes the decision
-surface smooth, while a high ``C`` aims at classifying all training examples
-correctly.  ``gamma`` defines how much influence a single training example has.
-The larger ``gamma`` is, the closer other examples must be to be affected.
+    >>> gram_test = np.dot(X_test, X_train.T)
+    >>> clf.predict(gram_test)
+    array([0, 1, 0])
 
-Proper choice of ``C`` and ``gamma`` is critical to the SVM's performance.  One
-is advised to use :class:`sklearn.model_selection.GridSearchCV` with 
-``C`` and ``gamma`` spaced exponentially far apart to choose good values.
-
-.. topic:: Examples:
-
- * :ref:`sphx_glr_auto_examples_svm_plot_rbf_parameters.py`
 
 .. _svm_mathematical_formulation:
 
 Mathematical formulation
 ========================
 
-A support vector machine constructs a hyper-plane or set of hyper-planes
-in a high or infinite dimensional space, which can be used for
+A support vector machine constructs a hyper-plane or set of hyper-planes in a
+high or infinite dimensional space, which can be used for
 classification, regression or other tasks. Intuitively, a good
 separation is achieved by the hyper-plane that has the largest distance
 to the nearest training data points of any class (so-called functional
 margin), since in general the larger the margin the lower the
-generalization error of the classifier.
-
+generalization error of the classifier. The figure below shows the decision
+function for a separable problem, with three samples within the margin
```

Comment:
```suggestion
function for a linearly separable problem, with three samples on the margin
```
maybe?

Reply:
indeed these are on the margin but in general, i.e. when classes aren't separable, the SVs are "within" the margin boundary